### ISSUE TITLE
Request for Data Output for Brain Surface Rendering

### ISSUE THREAD

**Issue by KGZaker** [ucl-pond/pySuStaIn#62]
Thank you so much for providing such a useful package! SuStaIn has been incredibly helpful. I noticed that the tutorial demonstrates how to plot heatmaps for each subtype and stage, but I would really appreciate the ability to generate similar plots (as shown below) that offer a more intuitive visualization of different brain regions for each subtype and stage.
![Arc 2024-10-21 12 37 17](https://github.com/user-attachments/assets/4879aea7-188f-448c-a28b-b7beec343c18)

From my understanding, it seems that pySuStaIn can currently only output heatmaps (via `pySuStaIn.ZscoreSustain._plot_sustain_model`). I’d like to inquire how the input data for brain surface rendering above is generated in this tutorial. Specifically, is the data derived by first assigning subjects to subtypes and stages, and then calculating the average z-score for each brain region across all subjects within each subtype and stage to serve as input for visualization?

If this is not the case, I assume the data used for brain rendering is the same as that used in the default heatmap generated by pySuStaIn. In this context, I’d like to ask whether you are considering adding a feature that allows for the export of the data used by `pySuStaIn.ZscoreSustain._plot_sustain_model` for visualization, rather than directly outputting the heatmap. This would allow us to perform our own brain surface renderings based on the exported data.

Thanks in advance for your kind help!

**Comment by noxtoby** [comment]
Those brains are generated by [BrainPainter](https://github.com/razvanmarinescu/brain-coloring) using a cumulative sum along the rows of `samples_sequence` (in the tutorial).

BrainPainter currently has a Google Colab web app [here](https://colab.research.google.com/drive/1qazM2eKGj18eXmu_IVH8HaAWyR9c8IY5#scrollTo=BPmGtEN6zI7O).

Enjoy!

**Comment by KGZaker** [comment]
@noxtoby Thank you very much for your reply! I tried using BrainPainter, and it’s really impressive. However, since our data is based on the Schaefer atlas, which doesn’t seem to be supported by BrainPainter, we may need to explore alternative visualization tools.

I appreciate your clarification regarding the use of a “cumulative sum along the rows of the samples_sequence” for visualization. However, I found the `samples_sequence` seems to represent the events occurring at each stage for every iteration. Specifically, events 0-4 correspond to biomarkers 1-5 reaching the z=1 threshold, events 5-9 correspond to the same biomarkers reaching z=2, and events 10-14 correspond to them reaching z=3 based on my understanding. Here is an example of the structure from the tutorial:
```python
>>> samples_sequence
array([[[ 2.,  2.,  2., ...,  2.,  2.,  2.],
        [ 7.,  7.,  7., ...,  7.,  7.,  7.],
        [ 0.,  0.,  0., ...,  0.,  0.,  0.],
        ...,
        [11., 11., 11., ..., 11., 11., 11.],
        [ 9.,  9.,  9., ...,  9.,  9.,  9.],
        [14., 14., 14., ..., 14., 14., 14.]],

       [[ 0.,  0.,  0., ...,  0.,  0.,  0.],
        [ 1.,  1.,  1., ...,  1.,  1.,  1.],
        [ 5.,  5.,  5., ...,  5.,  5.,  5.],
        ...,
        [12., 12., 12., ..., 12., 12., 12.],
        [11., 11., 11., ..., 11., 11., 11.],
        [13., 13., 13., ..., 13., 13., 13.]]])

>>> samples_sequence[0].shape
(15, 10000)
```

Based on this, I suppose the cumulative sum you’re referring to might be related to the `confusion matrix` derived from the `samples_sequence`, as generated in the `pySuStaIn.ZscoreSustain._plot_sustain_model` function.

However, after calculating the cumulative sum along the rows of the `confusion matrix`, I noticed that sometimes the same biomarker corresponds to different events within the same stage. For instance, in Column 3 (which I suppose represents stage 4), I observe a cumulative sum of 1 in Row 0 (biomarker 0 reaching z=1), a cumulative sum of 0.0762 in Row 5 (biomarker 0 reaching z=2), and a cumulative sum of 0 in Row 10 (biomarker 0 reaching z=3).
![Positron 2024-10-23 17 51 07](https://github.com/user-attachments/assets/22eb61c4-d6eb-4583-aab5-a23102935203)

In this case, how should I interpret the cumulative sum for biomarker 0 at stage 4? Should I simply select the highest cumulative sum, or should I treat the z thresholds as weights and calculate a weighted cumulative sum, such as `1*1 + 0.0762*2 + 0*3`?

I would greatly appreciate your insight on this.
